### ISSUE TITLE
Return output probabilities with Generate function 

### ISSUE THREAD

**Issue by tshrjn** [huggingface/transformers#8656]
# 🚀 Feature request

Output the token probabilities along with the tokens when generating sequence.

## Motivation

For understanding model confidence, this is quite useful. 
Also, for abstractive QA with long contexts, one needs to use doc-strides to take into account the contexts & then choose the best answer according to the probability of the generated text.

## Your contribution

I can try submitting a PR for non-beam decoding, but guidance would be appreciated.
Also, are there any existing solutions to this issue? If so, what & where?

**Comment by patrickvonplaten** [comment]
Duplicate of https://github.com/huggingface/transformers/issues/7654 . But yes, it seems like many people are asking for this feature and it should be quite straight-forward to implement it... -> feel free to give a shot :-)

**Comment by miguelvictor** [comment]
Hi. When can we expect this to be released? 😊

**Comment by patrickvonplaten** [comment]
Next week Tuesday or Wednesday :-)

**Comment by moqingyan** [comment]
Hey! Can anyone point me to the API/code location of this feature? Sorry if I have missed something. :) Thank you!

**Comment by miguelvictor** [comment]
@moqingyan https://huggingface.co/transformers/internal/generation_utils.html

**Comment by moqingyan** [comment]
> @moqingyan https://huggingface.co/transformers/internal/generation_utils.html

Thank you!

My problem was: I set the configuration `output_scores` to `True` in the `generate` function and failed obtain the scores in the returned results. After I struggled in the libraries for hours, I finally figured out I also need to set `return_dict_in_generate` to `True` to obtain the attention values from the `generate` function. :) 

I think this behavior is non-intuitive, as I have specified I need the scores in the output already. But anyway I figured this out. Hope this comment is helpful to anyone who runs into this issue.
